src/Wallabag/CoreBundle/Repository/EntryRepository.php

```diff
@@ -179,4 +180,53 @@ public function findOneByUsernameAndNotArchived($username)
             ->getQuery()
             ->getSingleResult();
     }
+
+    /**
+     * Remove a tag from all user entries.
+     *
+     * We need to loop on each entry attached to the given tag to remove it, since Doctrine doesn't know EntryTag entity because it's a ManyToMany relation.
+     * It could be faster with one query but I don't know how to retrieve the table name `entry_tag` which can have a prefix.
+     *
+     * @param int $userId
+     * @param Tag $tag
+     */
+    public function removeTag($userId, Tag $tag)
+    {
+        $entries = $this->getBuilderByUser($userId)
+            ->innerJoin('e.tags', 't')
+            ->andWhere('t.id = :tagId')->setParameter('tagId', $tag->getId())
+            ->getQuery()
+            ->getResult();
+
+        foreach ($entries as $entry) {
+            $entry->removeTag($tag);
+        }
+
+        $this->getEntityManager()->flush();
+
+        // An other solution can be to use raw query but I can't find a way to retrieve the `entry_tag` table name since it can be prefixed....
```

Comment:
Maybe the php doc before the function is necessary and we can remove these lines?